pkg/e2e/verify/samesite_cookie.go

```diff
@@ -16,28 +17,72 @@ const (
 	samesiteSetting = "Strict"
 	monNamespace    = "openshift-monitoring"
 	conNamespace    = "openshift-console"
+	supportVersion  = 46 // samesite cookie is only supported on >= v4.6.x
 )
 
-var samesiteTestName string = "[Suite: e2e] [OSD] Samesite Cookie Strict"
+var samesiteTestName string = "[Suite: informing] [OSD] Samesite Cookie Strict"
 
 var _ = ginkgo.Describe(samesiteTestName, func() {
 	h := helper.New()
 
 	ginkgo.Context("Validating samesite cookie", func() {
+
+		checkVersion := verifyVersion(h)
+
 		ginkgo.It("should be set for openshift-monitoring OSD managed routes", func() {
+			if checkVersion() {
+				ginkgo.Skip("skipping due to unsupported cluster version. Must be >=4.6.0")
+			}
 			foundKey, err := managedRoutes(h, monNamespace)
 			Expect(err).NotTo(HaveOccurred(), "failed getting routes for %v", monNamespace)
 			Expect(foundKey).Should(BeTrue(), "%v namespace routes have samesite cookie set", monNamespace)
 		}, 5)
 
 		ginkgo.It("should be set for openshift-console OSD managed routes", func() {
+			if checkVersion() {
+				ginkgo.Skip("skipping due to unsupported cluster version. Must be >=4.6.0")
+			}
 			foundKey, err := managedRoutes(h, conNamespace)
 			Expect(err).NotTo(HaveOccurred(), "failed getting routes for %v", conNamespace)
 			Expect(foundKey).Should(BeTrue(), "%v namespace routes have samesite cookie set", conNamespace)
 		}, 5)
 	})
 })
 
+func verifyVersion(h *helper.H) func() bool {
+	return func() bool {
+		clusterVersionObj, err := h.GetClusterVersion()
+		Expect(err).NotTo(HaveOccurred(), "failed getting cluster version")
+		Expect(clusterVersionObj).NotTo(BeNil())
+
+		// Get the cluster version and slice it, then convert the major/minor version to int Ex. majMinVersion := 46
+		splitVersion := strings.Split(clusterVersionObj.Status.Desired.Version, ".")
+
+		// Assume the len is <= 3 since semver major/minor could be something like 4.11.x
+		// Even if the semver maj/min exceed 4.9, this will still work. Ex. 4.11 = 411, which 411 > 46
+		if len(splitVersion) <= 1 || len(splitVersion) > 3 {
+			return true
+		} else {
+			if len(splitVersion) == 2 {
+				majMinVersion, err := strconv.Atoi(splitVersion[0] + splitVersion[1])
+				Expect(err).NotTo(HaveOccurred(), "failed normalizing major/minor version %v to integer %v", majMinVersion, err)
+				if majMinVersion < supportVersion {
+					return true
+				}
+			}
+
+			if len(splitVersion) == 3 {
+				majMinVersion, err := strconv.Atoi(splitVersion[0] + splitVersion[1] + splitVersion[2])
+				Expect(err).NotTo(HaveOccurred(), "failed normalizing major/minor version %v to integer %v", majMinVersion, err)
+				if majMinVersion < supportVersion {
+					return true
+				}
```

Comment:
So if `supportVersion` is `46`, and we have a cluster at version `4.7.11`, we end up comparing `4711 < 46`. Is that the intention? In general, integer comparison here might be problematic once the minor/patch versions have multiple digits? I don't know the constraints that you're working under here, so perhaps there's a reason that you're not worried about that?

I guess I should ask: What is the goal of this function? What does it verify about the version? What does a true or false return value *mean*?

Reply:
SameSite cookie being enables on routes in only supported in OSD versions >4.6. So, I am trying to verify the version here. In cases for version 5.11 for example I want 511 < 46 to fail. That is the intention here. Comparing integers is the way I am able to use conditionals in this sense to verify versions. I cannot simply look for versions >4.6 because 5.11 won't pass. So, this is the only way I can think of doing it. If you have other suggestions, then I am all ears.

This is why I check the length to be either 2 or 3 to account for multiple digits in the minor version (5.11) for example. Also, keep in mind I am splitting off the patch version, because I do not care about that in this case.

Reply:
I actually do not need the conditional to check if len is 3, so I will remove that.

Reply:
I can clean it up to this:
`if len(splitVersion) == 2 {
			majMinVersion, err := strconv.Atoi(splitVersion[0] + splitVersion[1])
			Expect(err).NotTo(HaveOccurred(), "failed normalizing major/minor version %v to integer %v", majMinVersion, err)
			if majMinVersion < supportVersion {
				return true
			} else {
				return false
			}
		}`